dinput_hook/renderer_utils.cpp

```diff
@@ -0,0 +1,428 @@
+#include "renderer_utils.h"
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+#include <string>
+#include <windows.h>
+#include <cmath>
+
+#include "globals.h"
+#include "types.h"
+
+extern "C" {
+#include <Primitives/rdMatrix.h>
+}
+
+extern "C" FILE *hook_log;
+
+rdMatrix44 g_projectionMatrix;
+rdMatrix44 g_modelViewMatrix;
+bool g_useAlphaMask;
+bool g_useFog;
+float g_fogColor[4];
+float g_fogStart;
+float g_fogEnd;
+
+extern "C" __declspec(dllexport) void renderer_setOrtho(rdMatrix44 *m, float left, float right,
+                                                        float bottom, float top, float nearVal,
+                                                        float farVal) {
+    float tx = -(right + left) / (right - left);
+    float ty = -(top + bottom) / (top - bottom);
+    float tz = -(farVal + nearVal) / (farVal - nearVal);
+
+    memcpy(m,
+           (float[16]){2.0f / (right - left), 0, 0, 0, 0, 2 / (top - bottom), 0, 0, 0, 0,
+                       -2 / (farVal - nearVal), 0, tx, ty, tz, 1},
+           sizeof(*m));
+}
+
+extern "C" __declspec(dllexport) void renderer_setProjectionMatrix(rdMatrix44 *projectionMatrix) {
+    memcpy(&g_projectionMatrix, projectionMatrix, sizeof(g_projectionMatrix));
```

Comment:
g_projectionMatrix = projectionMatrix

Reply:
Here I prefer keeping the memcpy as more explicit. I feel It can be confused too easily with a pointer copy (happened already haha)